### PR TITLE
Bug 1815765 - Add tab auto close banner to Tabs Tray rewrite

### PR DESCRIPTION
**Pull request by MozillaNoah**
[Figma for the `Banner`](https://www.figma.com/file/ujdiReypSypu2GhRgzArea/Android-Components?type=design&node-id=3554-35071&mode=design&t=twh3k6zcS1uzd2S8-0)

image

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1845786
https://bugzilla.mozilla.org/show_bug.cgi?id=1815765

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabAutoCloseBanner.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    onDismiss: () -> Unit,
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .background(FirefoxTheme.colors.layer1)
+            .padding(start = 11.dp, top = 21.dp, end = 16.dp),
+    ) {
+        Text(
+            text = stringResource(id = R.string.tab_tray_close_tabs_banner_message),
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabAutoCloseBanner.kt:44]: I'm confirming the real text style with Design. Won't land this until I hear back.

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabAutoCloseBanner.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Top-level UI for displaying the auto close CFR banner.
+ *
+ * @param onViewOptionsClick Invoked when the user clicks to view the auto close options.
+ * @param onDismiss Invoked when the user clicks to dismiss the banner.
+ */
+@Composable
+fun TabAutoCloseBanner(
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabAutoCloseBanner.kt:33]: the layout for this is derived from [`info_banner`](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/res/layout/info_banner.xml).

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt [hunk 3]
```diff
@@ -291,6 +295,16 @@ class TabsTrayFragment : AppCompatDialogFragment() {
...
+                        onTabAutoCloseBannerViewOptionsClick = {
+                            navigationInteractor.onTabSettingsClicked()
+                            requireContext().settings().shouldShowAutoCloseTabsBanner = false
+                        },
+                        onTabAutoCloseBannerDismiss = {
+                            requireContext().settings().shouldShowAutoCloseTabsBanner = false
+                        },
+                        onTabAutoCloseBannerShown = {
+                            requireContext().settings().lastCfrShownTimeInMillis = System.currentTimeMillis()
+                        },
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt:307]: I'll be writing a follow-up for the [interactor meta](https://bugzilla.mozilla.org/show_bug.cgi?id=1814225) to refactor this to the controller

**Comment by MozillaNoah** [comment]
Update: After speaking with Design, this PR is going to be retooled to use a `Banner` composable derived from the Banner design system component. So hold off on reviews until that work is done.

**Comment by MozillaNoah** [comment]
A top level `Banner` composable has been created and UX also asked for a divider line between the normal tabs tray banner UI and the auto close message banner. The auto close banner logic is now inside of `TabsTrayBanner` rather than inside of in the root `TabsTray`

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt [hunk 4]
```diff
@@ -57,6 +57,8 @@ import mozilla.components.browser.storage.sync.Tab as SyncTab
  * @param tabsTrayStore [TabsTrayStore] used to listen for changes to [TabsTrayState].
  * @param displayTabsInGrid Whether the normal and private tabs should be displayed in a grid.
  * @param isInDebugMode True for debug variant or if secret menu is enabled for this session.
+ * @param shouldShowTabAutoCloseBanner Whether the tab auto closer banner should be displayed.
+ * @param shouldShowInactiveTabsAutoCloseDialog Whether the inactive tabs auto close dialog should be displayed.
```
Comment (Alexandru2909) [inline fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt:61]: nit: We could also add the missing documentation for `onTabPageClick`.

**Review by Alexandru2909 — APPROVED** [review]
LGTM!

**Pull request merged**